templates/challenge.yaml

```
@@ -0,0 +1,60 @@
+# These are ACL lists that correlate to actual permissions
+# on synapse
+view_acl: &view
+  - READ
+
+download_acl: &download
+  - READ
+  - DOWNLOAD
+
+edit_acl: &edit
+  - READ
+  - DOWNLOAD
+  - UPDATE
+  - CREATE
+
+edit_and_delete_acl: &edit_and_delete
+  - READ
+  - DOWNLOAD
+  - UPDATE
+  - CREATE
+  - DELETE
+
+admin_acl: &admin
+  - READ
+  - DOWNLOAD
+  - UPDATE
+  - CREATE
+  - DELETE
+  - CHANGE_SETTINGS
+  - MODERATE
+  - CHANGE_PERMISSIONS
+
+StagingProject:
+  name: Challenge Staging Example
+  type: Project
+  acl:
+    # principal_id can be both user or team id
+    - principal_id: teamA
+      access_type: *admin
+    - principal_id: teamB
+      access_type: *view
+  children:
+    - name: Infrastructure
+      type: Folder
+
+LiveProject:
+  name: Challenge Live Example
+  type: Project
+  acl:
+    - principal_id: teamA
+      access_type: *admin
+    - principal_id: teamB
+      access_type: *view
+  children:
+    - name: Data
+      type: Folder
+    - name: Logs
+      type: Folder
+    - name: Resources
+      type: Folder
```

In this template, @thomasyu888 nested each top-level project under a key (_e.g._ `LiveProject`, `StagingProject`). I wonder if we should use the same format for the `children` attribute for consistency. Alternatively, we could make the top-level elements an array in YAML, but I prefer mappings over arrays (at least aesthetically). 

This isn't a hill I'll die on, but I would like to discuss why we would go one way or another. 

In the suggestion below, I'm taking advantage of the shortcut where the key is used as the folder name if `name` isn't given. The point of this is to make the value of `children` look like the root of the YAML template. 

```suggestion
  children:
    Data:
      type: Folder
    Logs:
      type: Folder
    Resources:
      type: Folder
```

---

I like this idea, except the short hand for this is already just:

```
children:
   - Data
   - Logs
```